packages/rolldown/tests/fixtures/output/es-module/common-chunks/_config.ts

```diff
@@ -0,0 +1,26 @@
+import type { RolldownOutputChunk } from 'rolldown'
+import { defineTest } from '@tests'
+import { expect } from 'vitest'
+
+const footerTxt = '// footer test\n'
+
+export default defineTest({
+  config: {
+    output: {
+      exports: 'named',
+      format: 'cjs',
+      esModule: true,
+    },
+  },
+  afterTest: (output) => {
+    expect(
+      output.output
+        .filter(({ type }) => type === 'chunk')
+        .every((chunk) =>
+          (chunk as RolldownOutputChunk).code.includes(
```

Comment:
The common chunks shouldn't has the `__esModule` flag, see [rollup example](https://rollupjs.org/repl/?version=4.20.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCgnLiUyRmZpbGUnKS50aGVuKChpbXBvcnRlZCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwYXNzZXJ0LnN0cmljdEVxdWFsKGltcG9ydGVkLnNoYXJlZCUyQyUyMCdzaGFyZWQnKSU1Q24lN0QpJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMG1haW4lMjAlM0QlMjAnbWFpbiclMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNvbnN0JTIwc2hhcmVkJTIwJTNEJTIwJ3NoYXJlZCclNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTJDJTIybmFtZSUyMiUzQSUyMmZpbGUuanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZXNNb2R1bGUlMjIlM0F0cnVlJTJDJTIyZXhwb3J0cyUyMiUzQSUyMm5hbWVkJTIyJTJDJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTdEJTJDJTIydHJlZXNoYWtlJTIyJTNBdHJ1ZSU3RCU3RA==)

Reply:
Got it.

Reply:
It should be a bug in handling the dynamic import already appearing in the main branch, which infers the common chunk as an entry point and removed all of exports.